R/clean_redcap_long.R

```diff
@@ -399,17 +398,37 @@ distill_repeat_table_long <- function(form_name,
 
 convert_mixed_instrument <- function(db_data_long, mixed_structure_ref) {
   for (i in seq_len(nrow(mixed_structure_ref))) {
-    field <- mixed_structure_ref$field_name[i]
-    form <- mixed_structure_ref$form_name[i]
-
-    # Create a logical mask for rows needing update
-    update_mask <- is.na(db_data_long$redcap_repeat_instance) & !is.na(db_data_long[[field]])
+    field <- mixed_structure_ref$field_name[i] # nolint: object_usage_linter
+    form <- mixed_structure_ref$form_name[i] # nolint: object_usage_linter
 
-    # Update redcap_repeat_instance
-    db_data_long$redcap_repeat_instance <- if_else(update_mask, 1, db_data_long$redcap_repeat_instance)
+    # Create an update mask column to identify which mixed structure rows need updates
+    db_data_long <- db_data_long %>%
+      mutate(
+        update_mask = case_when(
+          # repeat separately instances
+          !is.na({{ field }}) & is.na(.data$redcap_repeat_instance) ~ TRUE,
+          # repeat together instances
+          !is.na({{ field }}) & !is.na(.data$redcap_repeat_instance) & is.na(.data$redcap_repeat_instrument) ~ TRUE,
+          .default = FALSE
```

Comment:
For backwards compatibility. `.default` wasn't introduced until `dplyr 1.1.0`
```suggestion
          TRUE ~ FALSE
```

Reply:
Ah yes, good catch. I just hate "TRUE ~ *" that I started to block it from my memory.